src/game/NPCHandler.cpp

```diff
@@ -728,7 +756,17 @@ void WorldSession::HandleStableSwapPet(WorldPacket& recv_data)
 
     Pet* pet = _player->GetPet();
 
-    if (!pet || pet->getPetType() != HUNTER_PET)
+    if (!pet) // if pet not by your side, load from DB for a moment
```

Comment:
well, as long as it's unsummoned again I don't see an issue with it, if it's never visually displayed to the player.